vador/src/main/java/com/salesforce/vador/execution/strategies/AnnotationProcessor.kt

```diff
@@ -24,27 +35,120 @@ object AnnotationProcessorBase {
     val map = mapOfAnnotation?._1()
     val none = mapOfAnnotation?._2()
     return fields.mapNotNull { field ->
-      if (field.isAnnotationPresent(ValidateWith::class.java)) {
-        val annotation = field.getAnnotation(ValidateWith::class.java)
-        val failureKey = annotation.failureKey
-        if (ValidatorAnnotation1::class.java.isAssignableFrom(annotation.validator.java)) {
-          val validatorAnnotation =
-            annotation.validator.java.getDeclaredConstructor().newInstance()
-              as ValidatorAnnotation1<Any, FailureT?>
-          val validator =
-            Validator<ValidatableT?, FailureT?> {
-              validatorAnnotation.validate(
-                field,
-                FieldUtils.readField(field, bean, true) as Any,
-                map?.get(failureKey),
-                none,
+      if (field.annotations.isNotEmpty()) {
+        val validator: Validator<ValidatableT?, FailureT?>
+        when (field.annotations[0].annotationClass.simpleName) {
```

Comment:
We should compare with class itself, that we don't trip if it clashes with a custom validator. You can use `when` statement this way to compare class:

``` kotlin
when(field.annotations[0].annotationClass) {
		is MaxForInt -> ...
}
```